src/python_testing/matter_testing_infrastructure/matter/testing/spec_parsing.py

```diff
@@ -854,45 +871,118 @@ def parse_single_device_type(root: ElementTree.Element) -> tuple[dict[int, XmlDe
                     problems.append(ProblemNotice("Parse Device Type XML", location=location,
                                     severity=ProblemSeverity.WARNING, problem=f"Unknown cluster id {c.attrib['id']}"))
                     continue
+                # Workaround for 1.3 device types with zigbee clusters and old scenes
+                # This is OK because there are other tests that ensure that unknown clusters do not appear on the device
+                if cid not in cluster_definition_xml:
+                    logging.info(f"Skipping unknown cluster {cid:04X}")
+                    continue
                 conformance_xml, tmp_problem = get_conformance(c, cid)
                 if tmp_problem:
                     problems.append(tmp_problem)
-                conformance = parse_device_type_callable_from_xml(conformance_xml)
+                    continue
+                cluster_conformance_params = ConformanceParseParameters(
+                    feature_map=cluster_definition_xml[cid].feature_map, attribute_map=cluster_definition_xml[cid].attribute_map, command_map=cluster_definition_xml[cid].command_map)
+                conformance = parse_callable_from_xml(conformance_xml, cluster_conformance_params)
                 side_dict = {'server': ClusterSide.SERVER, 'client': ClusterSide.CLIENT}
                 side = side_dict[c.attrib['side']]
                 name = c.attrib['name']
                 if cid in CLUSTER_NAME_FIXES:
                     name = CLUSTER_NAME_FIXES[cid]
                 cluster = XmlDeviceTypeClusterRequirements(name=name, side=side, conformance=conformance)
+
+                def append_overrides(override_element_type: str):
+                    if override_element_type == 'feature':
+                        # The device types use feature name rather than feature code. So we need to build a new map.
+                        map = {f.name: id for id, f in cluster_definition_xml[cid].features.items()}
```

Comment:
This is shadowing the python `map` function. Suggest instead `feature_to_id_map`

Reply:
done, here and below. name_to_id_map